src/Symfony/Component/DependencyInjection/Attribute/AutowireInline.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Loader\Configurator\AbstractConfigurator;
+use Symfony\Component\DependencyInjection\Loader\Configurator\InlineServiceConfigurator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Attribute to tell a parameter to inline a service with custom parameters.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireInline extends Autowire implements BuildDefinitionInterface
+{
+    /**
+     * @param class-string|null $class
+     * @param array             $params Static parameters to pass to instance creation
+     * @param bool|class-string $lazy   Whether to use lazy-loading for this argument
+     */
+    public function __construct(
+        private readonly ?string $class = null,
```

Comment:
In `CheckDefinitionValidityPass`  we ask ppl to give a class even when a factory is used so we should make this required